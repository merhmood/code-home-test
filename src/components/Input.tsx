import React from "react";
import InputProps from "../types/InputProps";

const Input = ({
  placeholder,
  inputValue,
  inputType,
  onChange
}: InputProps) => {
  return (
    <input
      className="input"
      placeholder={placeholder}
      type={inputType}
      value={inputValue}
      onChange={onChange}
    />
  );
};

export default Input;
