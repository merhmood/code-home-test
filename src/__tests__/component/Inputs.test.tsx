import React from "react";
import renderer from "react-test-renderer";
import Input from "../../components/Input";

test("should render input component with fake props", () => {
  const onChangeStub = jest.fn();
  const fakeDataProps = {
    placeholder: "username",
    inputValue: "someRandomUserName",
    inputType: "text",
    onChange: onChangeStub
  };
  const tree = renderer.create(<Input {...fakeDataProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
