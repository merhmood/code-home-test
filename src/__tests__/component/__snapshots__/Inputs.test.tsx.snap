// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render input component with fake props 1`] = `
<input
  className="input"
  onChange={[MockFunction]}
  placeholder="username"
  type="text"
  value="someRandomUserName"
/>
`;
